CMakeLists.txt

```diff
@@ -2,6 +2,13 @@ cmake_minimum_required(VERSION 3.5)
 include($ENV{IDF_PATH}/tools/cmake/project.cmake)
 
 set(PROJECT_NAME "hackerhotel2024")
-set(PROJECT_VER "0.0.2")
+
+if(DEFINED ENV{PROJECT_VER})
+    # Strip any preceeding "v" from version (e.g. v1.1.1 > 1.1.1)
+    message(INFO "Setting Project version!")
+    string(REGEX REPLACE "v" "" PROJECT_VER "$ENV{PROJECT_VER}")
+else()
+    set(PROJECT_VER "999.999.999")
```

Comment:
Can we change this to "unknown"?

Reply:
```
#!/usr/bin/env bash

VERSION="$(git describe --dirty --always --tags 2> /dev/null)"

if [ -z "$VERSION" ]; then
    VERSION="unknown"
fi

VERSION_LENGTH=${#VERSION}
echo "add_compile_definitions(DEVICE_SW_VERSION=\"$VERSION\")"
echo "add_compile_definitions(DEVICE_SW_VERSION_LEN=$VERSION_LENGTH)"

```

Reply:
Maybe the script above can help? Takes the tag name and adds the commit hash if not on the tag itself, adds -dirty too if built without committing


tagname
tagname-shorthash
tagname-shorthash-dirty

depending on state